clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
Strictly speaking this patch doesn't change the behavior since the behavior is already "lost", right? The purpose of this patch is to make sure that the records don't get lost before a bad record.

I actually think the current behavior of skipping the bad record after exception is thrown should be the right behavior going forward. I also think it is backward compatible to keep this behavior since we don't have explicit documentation asking user to seek beyond bad record upon exception. What do you think?